File: codes/ruby/chapter05/answer_A27.rb
def gcd(a, b)
	while a >= 1 and b >= 1
		if a >= b
			a = a % b # a の値を変更する場合
		else
			b = b % a # b の値を変更する場合
		end
	end
	if a >= 1
		return a
	else
		return b
	end
end

# 入力
a, b = gets.split.map(&:to_i)
puts gcd(a, b)

File: codes/elixir/chapter05/answer_A27.exs
defmodule Gcd do
  defmodule Math do
    def gcd(a, b) when a >= 1 and b >= 1, do: gcd(a, b, min(a, b))

    defp gcd(a, _b, 0), do: a
    defp gcd(a, b, min) when rem(a, min) == 0 and rem(b, min) == 0, do: min
    defp gcd(a, b, min), do: gcd(a, b, min - 1)
  end
end

a = String.trim(IO.gets("")) |> String.to_integer
b = String.trim(IO.gets("")) |> String.to_integer
IO.puts(Gcd.Math.gcd(a, b))
